xcube/core/tile.py

```diff
@@ -404,6 +383,6 @@ def to_datetime(datetime_str: str, dim_var: xr.DataArray):
                                  f' value for dimension {dim!r}') from e
 
     if var is not None:
-        return _ensure_time_compatible(var, parsed_labels)
+        return ensure_time_label_compatible(var, parsed_labels)
```

Comment:
`xcube/core/tile.py` will be deprecated. For xcube server NG, a lot of code changes have been introduced. After merging this PR, please update also https://github.com/dcs4cop/xcube/pull/677.